### ISSUE TITLE
Feature Request: Add NTFS Reparse Point Target (Symlink/Junction/Hardlink) Column

### ISSUE THREAD

**Issue by CollinChaffin** [tablacus/TablacusExplorerAddons#39]
Hi Gaku!

I have become a number one fan of Tablacus!  I wanted to open this feature request because you have accomplished adding the Directory Size (total) the way it should have always been but has been missing from every version of Explorer.  Along the same lines is the set of extended NTFS attributes such as reparse points (junctions/symbolic links etc.) which I see you already have coded to be detected perfectly using the "colored with attributes) which very quickly colors all my junctions.

So, since that code can easily color them, there are a couple other explorer replacements that can display indicators for junctions which would be better than nothing, but MS has always had an Explorer column titled "Link Target" that of course is NOT for junctions, but if Tablacus could add 2 columns with an addon like this as an example similar to how you implemented the "total file size" column:

EXAMPLE  `REPARSE TYPE: S=symlink/J=junction/H=hardlink`

| Name           | Size  | Type        | _REPARSE TYPE_ | _REPARSE TARGET_ | 
|----------------|-------|-------------|--------------|----------------|
| Data1          |       | File Folder |              |                | 
| Data2          |       | File Folder | J            | D:\Data        | 
| Data3          |       | File Folder | S            | E:\Data        | 
| ReparseFile.txt     | 100kb | Text File   | H            | D:\File01.txt  | 
| NormalFile.txt | 10kb  | Text File   |              |                | 

Adding not only the ability to quickly see a list indicating the type and more importantly the target of the links, but then utilize the built-in column sorting to be able to in a click see all junctions sorted to the top or sort by the link target would just be amazing.

Hopefully you can see the value in this and will consider adding it to Tablacus as I have seen many posts from people asking how to enable this in Windows Explorer and I would love to be able to give them yet one more reason to switch to Tablacus!  Thanks!

**Comment by tablacus** [comment]
Unfortunately, I don't know how to add new columns on windows vista or later.
I can only reuse exists columns like this.

Tablacus Explorer 18.11.1
![image](https://user-images.githubusercontent.com/5156977/47853746-0aba4000-de23-11e8-86c1-8e450905a9df.png)

Explorer
![image](https://user-images.githubusercontent.com/5156977/47853758-1c034c80-de23-11e8-8465-f5fad8337546.png)

Please wait because it is false detection.
https://github.com/tablacus/TablacusExplorer/issues/146#issuecomment-435033704

Best regards,

**Comment by CollinChaffin** [comment]
I'm confused but you DID already add a column - total file size which computes the total folder size which is NOT a default WIN7 explorer column.  So, since you have the code to detect the reparse points, I wondered why you cannot do what you did with "Total file size" column there too?

**Comment by tablacus** [comment]
"Total file size" is not a new column.
In fact, Explorer already has an column "Total file size".

![image](https://user-images.githubusercontent.com/5156977/47922704-0a8d7380-defb-11e8-8137-6d6ea04cd8e5.png)

Regards.

**Comment by CollinChaffin** [comment]
Correct, but through API calls you are populating it.  Windows Explorer has NEVER calculated total file size for folders - which is exactly what your picture above shows.  So, why not pick one of the hundred of other UNUSED columns that Explorer also does not even populate properly (such as the exact one I mentioned that like total file size even matches the basic description of what I'm asking - "Link Target") so I don't know why you are so against this?  I literally already use your code to COLOR my junctions so that code is even written this would be easy for you since you've done both parts already I would do it but I'd first have to reverse all your code since there are literally zero comments anywhere which makes it very difficult to understand and contribute.  Not to mention I'm a dotnet/C/CPP coder (been many years since I've had to do any C/CPP but did my share back in the day) but not Java/Javascript which makes the addons challenging for me.

Here is all you need to do to implement this feature that I guarantee would be WIDELY used by TE users once they knew it existed:

- `(Step 1) - Hook through your existing API functions to grab "Link Target"`  _This is exactly what you do to populate the otherwise BLANK "Total File Size" column._

- `(Step 2) - Use the same code you already have implemented to filter attributes of type JUNCTION` _This is easy no matter which language - **Colored with Attributes v1.02** is existing addon JS code that does it and I am pretty certain the main TE CPP code also has basic file attribute STRUCTS in use that can also do it.  Instead of coloring the lines, send that return output of attribute filter function as the input into to the STEP1 function to perform the column population.  NOTE: This is the EXACT SAME code function/path that is populating "Total File Size" today - EXACT!_

Make sense?  Back in the day, and even now in other languages I am very familiar and make use of the WIN32API and it's capabilities in my own code so I know this would not be terribly difficult with a grasp of the TFS code (which you have as the author).  Despite this being self-serving and me wanting it for myself, I am also trying to help make this THE goto file explorer replacement for everyone.  Not to mention, take this one new step and you have now literally put in place a cookie-cutter formula that can be re-used to add in other columns/data missing from Explorer!  

I would love to collaborate with you on this; I simply need some direction/assistance in your code and the current code path of the "Total File Size" start to finish.  Not only the initial CPP code, but whether there is (I believe so) also the JS addon code responsible for anything EXCEPT sorting.  I'm not yet concerned with sorting differences, we can assume the same restrictions are in place for this as TFS (Total File Size), cannot use W7explorer for detail view, might have to do an addon button for proper sorting/resorting of this new column, etc.

Can we give this a try together?  When you get a bit of time could you possibly add even just a few code comments to the relevant sections so it's easier for others to collaborate with you on this (and other issues/enhancements)?  Thanks!!



### EXAMPLE:  Junction Coloring + Total File Size population
## MORE IMPORTANTLY: Total File Size (TFS) column (normally BLANK) populated breaking further rules (via WIN32API) and even properly traverses then totals the NTFS Junction TARGET folder!:

![snag_11-3-2018_17-39-08-edited](https://user-images.githubusercontent.com/9301483/47958160-7fc17d00-dfbd-11e8-90c2-50711406b710.png)

**Comment by CollinChaffin** [comment]
Before you can even reply I have to say how AMAZING it is that you have totally implemented this in a matter of hours in the TE v18.11.3 update!

For those who truly appreciate what this enhancement means.....something no explorer has ever done until now in Windows (Microsoft, are you paying attention?):

![snag_11-3-2018_23-36-28-edited](https://user-images.githubusercontent.com/9301483/47960066-f83f3280-dfeb-11e8-91b2-a4728dc4496b.png)

This is exactly what I suggested!  Since there already is an indicator of the "L" for link in attributes, that other column I showed as an idea above was simply like grabbing the other unused column right next to the "link target" you used of "link status" and I wasn't sure if possible to then generate a bit more detail than the one "L" and add anything from a single letter to denote "J" for Junction, "S" for Symlink, etc. or even to spell it out since it's always really one type or the other.....but again that column was merely an optional extra to compliment this.  

This is huge, very huge that TE is really the only full Windows forms/GUI explorer replacement I have seen that can correctly resolve and display every reparse target in the file list display.  Thanks so much!
